File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
  let stack = [];
    
    for (let i = 0; i < s.length; i++) {
        if (stack.length !== 0 && stack[stack.length - 1] === s[i]) {
            // If the top of the stack is the same as the current character, pop from the stack
            stack.pop();
        } else {
            // If the stack is empty or the top of the stack is not the same as the current character, push onto the stack
            stack.push(s[i]);
        }
    }

    // If the stack is empty, return 1, else return 0
    return stack.length === 0 ? 1 : 0;
    
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = 0
    stackS = []
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.    
    
    
    for i in range(len(s)):
        
        #스택이 비어있는 경우
        if len(stackS) == 0:
            stackS.append(s[i])
        #제일 뒤에 있는 원소를 비교
        elif stackS[-1] == s[i]:
            stackS.pop()
        #삭제 가능한 문자가 없을 때
        else:
            stackS.append(s[i])
    
    #문자열 제거 후 남아있는게 없으면 1 반환
    if len(stackS) == 0:
        return 1
    
    return answer